Reconciliation section

Nathan-h05 opened the pull request:
## 🔧 Description
<!-- What changes does this PR introduce? -->
- Add testimonials carousel with an auth-only draft slide for creation.
- Add testimonials API integration (GET/save-all/upload-image/delete) and wire up frontend save/delete.
- Upload images on Save with client-side compression to avoid 10MB errors. (in ResizeImageFile)
- Modified EditableText.tsx (Quill.js) to allow for highlight and colored text. 
- Set up save, and delete with user feedback.

## ✅ Related Issues
N/A

## 📝 Improvements
- Refactor with backend and database, switched to its own table. 
- EditableText is more customizable with text colour and highlight.
- Split testimonials into dedicated types and endpoints, keeping SectionDataType minimal (text/image only).
- Ensured section save requests only send required fields, aligning with backend SectionModel.

## 🧪 Testing
- [x] Saving slide gives user feedback and saves to the database
- [x] Deleting slide gives user feedback and saves to the database
- [x] Skeleton slide only shows when logged in, and always the last slide
- [x] Image compression works with big files (Over 10MB)
- [x] Code runs locally

## 📸 Screenshots (if UI changes)
<img width="1907" height="850" alt="image" src="https://github.com/user-attachments/assets/14381f0c-7c39-4612-a6fa-ca4f20fef808" />


Logged In view:
(Font was set to white)
<img width="1906" height="846" alt="image" src="https://github.com/user-attachments/assets/4b0ee60a-b2ae-4d10-a9eb-350eb3b49c4b" />

<img width="1907" height="846" alt="image" src="https://github.com/user-attachments/assets/48c5c8fd-4e9c-439a-a955-f341476fd3ed" />


Skeleton slide:
<img width="1905" height="848" alt="image" src="https://github.com/user-attachments/assets/d6ea527c-1f52-4a7f-9e3a-4fcfab45085d" />

Highlight and colored text:
<img width="642" height="232" alt="image" src="https://github.com/user-attachments/assets/de058a30-bfef-4759-826d-752acd2a2a97" />

Feedback:
Saved success:
<img width="621" height="298" alt="image" src="https://github.com/user-attachments/assets/ffd83cb0-9ceb-4453-9879-26edd5c69de4" />

Delete slide success:
<img width="270" height="148" alt="image" src="https://github.com/user-attachments/assets/26022bce-d79e-4ade-be1a-ce69217c2e6c" />

Failed slide save:
<img width="354" height="199" alt="image" src="https://github.com/user-attachments/assets/b231bb97-6317-4b39-b809-3341c1cf90a8" />

Failed slide delete:
<img width="297" height="175" alt="image" src="https://github.com/user-attachments/assets/30eab472-a29f-4b39-b920-bd3293326d98" />


## 🚀 Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas

Nathan-h05 committed:
Basic slide system

Nathan-h05 committed:
Merge branch 'dev' into ReconciliationSection
For Text saving feature and  Image Saving Feature

Nathan-h05 committed:
added slides support for Reconciliation section

- Added /save-slides route + controller with multer multi-image handling
- Updated schema & shared types to include optional slides array
- Refactored ReconciliationSection: dynamic add/remove/edit slides + image file tracking
- Implemented FormData save logic (selective slide image uploads)
- NOTE: Remaining UI/UX polish & edge-case handling still pending

Nathan-h05 committed:
Merge remote-tracking branch 'origin/dev' into ReconciliationSection

Nathan-h05 committed:
Refactor testimonials to use Quill.js, skeleton draft, new collection

- Reworked EditableText (Quill.js), adding text and highlight color and optional placeholders
- Replace creation slide with a persistent skeleton draft as the creation flow
- Add testimonials API (GET, POST /save-all, POST /upload-image, DELETE /:id) on new “testimonials” collection
- Improve delete/save UX (handle last-slide, clamp index, success/error toasts)
- Added utils client-side compression for before Cloudinary upload (on Save) to avoid 10MB limit

Nathan-h05 committed:
code cleanup

JAlipes reviewed (changes requested):
Fall back should use the default text instead of empty strings so then text and images are always being displayed instead of empty sections being shown
<img width="1447" height="384" alt="image" src="https://github.com/user-attachments/assets/88b33614-aca3-4679-bf0c-ea9e7fccce10" />

For ReconcliationSection.tsx this was a miscommunication on my part but the title is fixed and will stay between each slides it shouldn't be editable
<img width="1364" height="332" alt="image" src="https://github.com/user-attachments/assets/5ad48362-e885-4702-b470-c474485fe2f5" />

For SaveEditsButton.tsx ensure type goes into the correct types file but I like the idea of customizing the buttons
<img width="435" height="233" alt="image" src="https://github.com/user-attachments/assets/c37a082f-0770-4d8f-9e1f-45b4a23eef6e" />

For the DeleteTestimonials.ts make sure to use the GetEnvOrFail function for security of empty vars. Prevents us from running the app with empty env vars
- Same with HandleGetTestimonials.ts
- HandleSaveTestimonials
- UploadTestimonialImage
<img width="1452" height="486" alt="image" src="https://github.com/user-attachments/assets/eaaf9045-5f19-4f72-95cd-0f374f1fc9b8" />


Instead of reusing SaveEditsbutton to display success or fail alerts we will be using Swal2. This keeps the functionality of the SaveEditsButton purposeful to saving instead of both saving and alerts.

Everything else looks good though nice catch on the sectionDataTypes making sure that we only receive and send information that's needed. You can add that as an improvement for the improvements section of the description

Nathan-h05 committed:
Addressed PR review changes

- Switched testimonial utils to GetEnvVarOrFail
- Moved SaveEditsButton props to shared types and removed success label handling
- ReconciliationSection now uses a single static title (no per-slide titles)

Nathan-h05 committed:
updated ReconciliationSection bg color to 'brandRed'

Nathan-h05 committed:
added defaultTestimonials for fallback

JAlipes reviewed (approved):
Looks Good
